discovery-provider/src/challenges/challenge_event_bus.py

```diff
@@ -0,0 +1,71 @@
+import json
+import enum
+import logging
+
+from collections import defaultdict
+
+logger = logging.getLogger(__name__)
+REDIS_QUEUE_PREFIX = 'challenges-event-queue'
+
+class ChallengeEvent(str, enum.Enum):
+    profile_update = 'profile_update'
+    repost = 'repost'
+    follow = 'follow'
+    favorite = 'favorite'
+
+class ChallengeEventBus:
```

Comment:
Would be really great if we could expose some sort of endpoint/health check for this guy to see what's in the queue, how it's progressing, etc.

Reply:
I really like this idea. I'm going to include this on the subsequent PR to avoid scope creep, but this is a good idea. We should handle this like we do the trending last compute timestamp